### PR TITLE
feat(ui): refactor theming classes

### PR DESCRIPTION
**Pull request by GroovinChip**
This PR refactors a number of theme classes in the following ways:
* Move all theme classes into their own files under a `/themes` directory, with the exception of `stream_chat_theme.dart`
* Add `InheritedTheme` classes to most themes
* Add `lerp` functions to all themes
* Add diagnosticable properties to all themes
* Add `==` and `hashCode` functions to all themes

Additionally, tests have been created for themes that did not previously have tests

## CLA

- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required).
- [x] The code changes follow best practices
- [x] Code changes are tested (add some information if not applicable)

### CODE REVIEW

File: packages/stream_chat_flutter/lib/src/channel_preview.dart [hunk 1]
```diff
@@ -183,8 +183,9 @@ class ChannelPreview extends StatelessWidget {
 
           return Text(
             stringDate,
-            style:
-                StreamChatTheme.of(context).channelPreviewTheme.lastMessageAt,
+            style: StreamChatTheme.of(context)
+                .channelPreviewTheme
+                .lastMessageAtStyle,
```
Comment (xsahil03x) [inline packages/stream_chat_flutter/lib/src/channel_preview.dart:188]: this

File: packages/stream_chat_flutter/lib/src/channel_bottom_sheet.dart [hunk 2]
```diff
@@ -65,7 +65,7 @@ class _ChannelBottomSheetState extends State<ChannelBottomSheet> {
                     showTypingIndicator: false,
                     channel: _streamChannelState.channel,
                     textStyle:
-                        _streamChatThemeData.channelPreviewTheme.subtitle,
+                        _streamChatThemeData.channelPreviewTheme.subtitleStyle,
```
Comment (xsahil03x) [inline packages/stream_chat_flutter/lib/src/channel_bottom_sheet.dart:68]: this

File: packages/stream_chat_flutter/lib/src/channel_list_header.dart [hunk 3]
```diff
@@ -248,7 +246,7 @@ class ChannelListHeader extends StatelessWidget implements PreferredSizeWidget {
       children: [
         Text(
           context.translations.offlineLabel,
-          style: chatThemeData.channelListHeaderTheme.title?.copyWith(
+          style: chatThemeData.channelListHeaderTheme.titleStyle?.copyWith(
```
Comment (xsahil03x) [inline packages/stream_chat_flutter/lib/src/channel_list_header.dart:249]: this

File: packages/stream_chat_flutter/lib/src/channel_list_header.dart [hunk 4]
```diff
@@ -259,7 +257,7 @@ class ChannelListHeader extends StatelessWidget implements PreferredSizeWidget {
             ..openConnection(),
           child: Text(
             context.translations.tryAgainLabel,
-            style: chatThemeData.channelListHeaderTheme.title?.copyWith(
+            style: chatThemeData.channelListHeaderTheme.titleStyle?.copyWith(
```
Comment (xsahil03x) [inline packages/stream_chat_flutter/lib/src/channel_list_header.dart:260]: this

File: packages/stream_chat_flutter/lib/src/channel_preview.dart [hunk 5]
```diff
@@ -200,15 +201,15 @@ class ChannelPreview extends StatelessWidget {
...
+            style: chatThemeData.channelPreviewTheme.subtitleStyle,
           ),
         ],
       );
     }
     return TypingIndicator(
       channel: channel,
       alternativeWidget: _buildLastMessage(context),
-      style: chatThemeData.channelPreviewTheme.subtitle,
+      style: chatThemeData.channelPreviewTheme.subtitleStyle,
```
Comment (xsahil03x) [inline packages/stream_chat_flutter/lib/src/channel_preview.dart:212]: this

File: packages/stream_chat_flutter/lib/src/full_screen_media.dart [hunk 6]
```diff
@@ -125,7 +125,6 @@ class _FullScreenMediaState extends State<FullScreenMedia>
                             backgroundDecoration: BoxDecoration(
                               color: ColorTween(
                                 begin: StreamChatTheme.of(context)
-                                    .channelTheme
                                     .channelHeaderTheme
                                     .color,
```
Comment (xsahil03x) [inline packages/stream_chat_flutter/lib/src/full_screen_media.dart:129]: this

File: packages/stream_chat_flutter/lib/src/message_actions_modal.dart [hunk 7]
```diff
@@ -565,7 +566,8 @@ class _MessageActionsModalState extends State<MessageActionsModal> {
       elevation: 2,
       clipBehavior: Clip.hardEdge,
       isScrollControlled: true,
-      backgroundColor: streamChatThemeData.messageInputTheme.inputBackground,
+      backgroundColor:
+          streamChatThemeData.messageInputTheme.inputBackgroundColor,
```
Comment (xsahil03x) [inline packages/stream_chat_flutter/lib/src/message_actions_modal.dart:570]: this

File: packages/stream_chat_flutter/lib/src/message_list_view.dart [hunk 8]
```diff
@@ -628,7 +629,7 @@ class _MessageListViewState extends State<MessageListView> {
         child: Text(
           context.translations.threadSeparatorText(replyCount),
           textAlign: TextAlign.center,
-          style: _streamTheme.channelTheme.channelHeaderTheme.subtitle,
+          style: _streamTheme.channelHeaderTheme.subtitleStyle,
```
Comment (xsahil03x) [inline packages/stream_chat_flutter/lib/src/message_list_view.dart:632]: this

File: packages/stream_chat_flutter/lib/src/message_search_item.dart [hunk 9]
```diff
@@ -94,7 +94,7 @@ class MessageSearchItem extends StatelessWidget {
 
     return Text(
       stringDate,
-      style: StreamChatTheme.of(context).channelPreviewTheme.lastMessageAt,
+      style: StreamChatTheme.of(context).channelPreviewTheme.lastMessageAtStyle,
```
Comment (xsahil03x) [inline packages/stream_chat_flutter/lib/src/message_search_item.dart:97]: this

File: packages/stream_chat_flutter/lib/src/message_search_item.dart [hunk 10]
```diff
@@ -128,12 +128,12 @@ class MessageSearchItem extends StatelessWidget {
         text!,
         message.mentionedUsers,
         message.attachments,
-        chatThemeData.channelPreviewTheme.subtitle?.copyWith(
+        chatThemeData.channelPreviewTheme.subtitleStyle?.copyWith(
```
Comment (xsahil03x) [inline packages/stream_chat_flutter/lib/src/message_search_item.dart:131]: this

File: packages/stream_chat_flutter/lib/src/channel_preview.dart (hunk 1 above)
Comment (GroovinChip) [reply packages/stream_chat_flutter/lib/src/channel_preview.dart:188]: Done

File: packages/stream_chat_flutter/lib/src/message_search_item.dart [hunk 11]
```diff
@@ -128,12 +128,12 @@ class MessageSearchItem extends StatelessWidget {
...
           fontStyle: (message.isSystem || message.isDeleted)
               ? FontStyle.italic
               : FontStyle.normal,
         ),
-        chatThemeData.channelPreviewTheme.subtitle?.copyWith(
+        chatThemeData.channelPreviewTheme.subtitleStyle?.copyWith(
```
Comment (xsahil03x) [inline packages/stream_chat_flutter/lib/src/message_search_item.dart:136]: this

File: packages/stream_chat_flutter/lib/src/channel_bottom_sheet.dart (hunk 2 above)
Comment (GroovinChip) [reply packages/stream_chat_flutter/lib/src/channel_bottom_sheet.dart:68]: done

File: packages/stream_chat_flutter/lib/src/channel_info.dart [hunk 12]
```diff
@@ -65,10 +65,7 @@ class ChannelInfo extends StatelessWidget {
       }
       alternativeWidget = Text(
         text,
-        style: StreamChatTheme.of(context)
-            .channelTheme
-            .channelHeaderTheme
-            .subtitle,
+        style: StreamChatTheme.of(context).channelHeaderTheme.subtitleStyle,
```
Comment (xsahil03x) [inline packages/stream_chat_flutter/lib/src/channel_info.dart:68]: this

File: packages/stream_chat_flutter/lib/src/channel_list_header.dart (hunk 3 above)
Comment (GroovinChip) [reply packages/stream_chat_flutter/lib/src/channel_list_header.dart:249]: already done

File: packages/stream_chat_flutter/lib/src/channel_header.dart [hunk 13]
```diff
@@ -169,16 +168,16 @@ class ChannelHeader extends StatelessWidget implements PreferredSizeWidget {
...
                         ),
                     const SizedBox(height: 2),
                     subtitle ??
                         ChannelInfo(
                           showTypingIndicator: showTypingIndicator,
                           channel: channel,
-                          textStyle: chatThemeData
-                              .channelTheme.channelHeaderTheme.subtitle,
+                          textStyle:
+                              chatThemeData.channelHeaderTheme.subtitleStyle,
```
Comment (xsahil03x) [inline packages/stream_chat_flutter/lib/src/channel_header.dart:180]: this

File: packages/stream_chat_flutter/lib/src/channel_list_header.dart (hunk 4 above)
Comment (GroovinChip) [reply packages/stream_chat_flutter/lib/src/channel_list_header.dart:260]: already done

File: packages/stream_chat_flutter/lib/src/channel_header.dart [hunk 14]
```diff
@@ -169,16 +168,16 @@ class ChannelHeader extends StatelessWidget implements PreferredSizeWidget {
                   children: <Widget>[
                     title ??
                         ChannelName(
-                          textStyle: chatThemeData
-                              .channelTheme.channelHeaderTheme.title,
+                          textStyle:
```
Comment (xsahil03x) [inline packages/stream_chat_flutter/lib/src/channel_header.dart:171]: this

File: packages/stream_chat_flutter/lib/src/channel_header.dart [hunk 15]
```diff
@@ -169,16 +168,16 @@ class ChannelHeader extends StatelessWidget implements PreferredSizeWidget {
...
+                              chatThemeData.channelHeaderTheme.titleStyle,
```
Comment (xsahil03x) [inline packages/stream_chat_flutter/lib/src/channel_header.dart:172]: this

File: packages/stream_chat_flutter/lib/src/channel_header.dart [hunk 16]
```diff
@@ -141,17 +141,16 @@ class ChannelHeader extends StatelessWidget implements PreferredSizeWidget {
...
-                chatThemeData.channelTheme.channelHeaderTheme.color,
+            backgroundColor: chatThemeData.channelHeaderTheme.color,
             actions: actions ??
                 <Widget>[
                   Padding(
                     padding: const EdgeInsets.only(right: 10),
                     child: Center(
                       child: ChannelAvatar(
-                        borderRadius: chatThemeData.channelTheme
+                        borderRadius: chatThemeData
```
Comment (xsahil03x) [inline packages/stream_chat_flutter/lib/src/channel_header.dart:151]: this

File: packages/stream_chat_flutter/lib/src/channel_header.dart [hunk 17]
```diff
@@ -141,17 +141,16 @@ class ChannelHeader extends StatelessWidget implements PreferredSizeWidget {
...
                             .channelHeaderTheme.avatarTheme?.borderRadius,
-                        constraints: chatThemeData.channelTheme
+                        constraints: chatThemeData
```
Comment (xsahil03x) [inline packages/stream_chat_flutter/lib/src/channel_header.dart:153]: this

File: packages/stream_chat_flutter/example/lib/tutorial_part_3.dart [hunk 18]
```diff
@@ -118,13 +118,15 @@ class ChannelListPage extends StatelessWidget {
...
-                ),
+        textStyle: StreamChatTheme.of(context)
+            .channelPreviewTheme
+            .titleStyle!
+            .copyWith(
+              color: StreamChatTheme.of(context)
+                  .colorTheme
+                  .textHighEmphasis
+                  .withOpacity(opacity),
+            ),
```
Comment (xsahil03x) [inline packages/stream_chat_flutter/example/lib/tutorial_part_3.dart:129]: this

File: packages/stream_chat_flutter/lib/src/channel_list_header.dart (hunk 3 above)
Comment (GroovinChip) [reply packages/stream_chat_flutter/lib/src/channel_list_header.dart:249]: Never mind, I see it. Done now

File: packages/stream_chat_flutter/lib/src/channel_list_header.dart (hunk 4 above)
Comment (GroovinChip) [reply packages/stream_chat_flutter/lib/src/channel_list_header.dart:260]: Never mind, done now

File: packages/stream_chat_flutter/lib/src/channel_preview.dart [hunk 19]
```diff
@@ -200,15 +201,15 @@ class ChannelPreview extends StatelessWidget {
           ),
           Text(
             '  ${context.translations.channelIsMutedText}',
-            style: chatThemeData.channelPreviewTheme.subtitle,
+            style: chatThemeData.channelPreviewTheme.subtitleStyle,
```
[the remark this answers is not in the document]
Comment (GroovinChip) [reply packages/stream_chat_flutter/lib/src/channel_preview.dart:204]: Done

File: packages/stream_chat_flutter/lib/src/channel_preview.dart (hunk 5 above)
Comment (GroovinChip) [reply packages/stream_chat_flutter/lib/src/channel_preview.dart:212]: Done

File: packages/stream_chat_flutter/lib/src/full_screen_media.dart (hunk 6 above)
Comment (GroovinChip) [reply packages/stream_chat_flutter/lib/src/full_screen_media.dart:129]: Done

File: packages/stream_chat_flutter/lib/src/message_list_view.dart (hunk 8 above)
Comment (GroovinChip) [reply packages/stream_chat_flutter/lib/src/message_list_view.dart:632]: Done

File: packages/stream_chat_flutter/lib/src/message_search_item.dart (hunk 9 above)
Comment (GroovinChip) [reply packages/stream_chat_flutter/lib/src/message_search_item.dart:97]: Done

File: packages/stream_chat_flutter/lib/src/message_actions_modal.dart (hunk 7 above)
Comment (GroovinChip) [reply packages/stream_chat_flutter/lib/src/message_actions_modal.dart:570]: Done

File: packages/stream_chat_flutter/lib/src/message_search_item.dart (hunk 10 above)
Comment (GroovinChip) [reply packages/stream_chat_flutter/lib/src/message_search_item.dart:131]: Done

File: packages/stream_chat_flutter/lib/src/message_search_item.dart (hunk 11 above)
Comment (GroovinChip) [reply packages/stream_chat_flutter/lib/src/message_search_item.dart:136]: Done

File: packages/stream_chat_flutter/lib/src/channel_info.dart (hunk 12 above)
Comment (GroovinChip) [reply packages/stream_chat_flutter/lib/src/channel_info.dart:68]: Done

File: packages/stream_chat_flutter/lib/src/channel_preview.dart [hunk 20]
```diff
@@ -249,13 +248,14 @@ class ChannelPreview extends StatelessWidget {
                 text,
                 lastMessage.mentionedUsers,
                 lastMessage.attachments,
-                chatThemeData.channelPreviewTheme.subtitle?.copyWith(
-                    color: chatThemeData.channelPreviewTheme.subtitle?.color,
+                chatThemeData.channelPreviewTheme.subtitleStyle?.copyWith(
+                    color:
+                        chatThemeData.channelPreviewTheme.subtitleStyle?.color,
```
Comment (xsahil03x) [inline packages/stream_chat_flutter/lib/src/channel_preview.dart:253]: this

File: packages/stream_chat_flutter/lib/src/channel_preview.dart [hunk 21]
```diff
@@ -249,13 +248,14 @@ class ChannelPreview extends StatelessWidget {
...
                     fontStyle: (lastMessage.isSystem || lastMessage.isDeleted)
                         ? FontStyle.italic
                         : FontStyle.normal),
-                chatThemeData.channelPreviewTheme.subtitle?.copyWith(
-                  color: chatThemeData.channelPreviewTheme.subtitle?.color,
+                chatThemeData.channelPreviewTheme.subtitleStyle?.copyWith(
+                  color: chatThemeData.channelPreviewTheme.subtitleStyle?.color,
```
Comment (xsahil03x) [inline packages/stream_chat_flutter/lib/src/channel_preview.dart:258]: this

Comment (xsahil03x) [reply packages/stream_chat_flutter/lib/src/channel_preview.dart]: fixed

File: packages/stream_chat_flutter/lib/src/channel_preview.dart (hunk 20 above)
Comment (xsahil03x) [reply packages/stream_chat_flutter/lib/src/channel_preview.dart]: fixed

**Review by xsahil03x — APPROVED** [review]
LGTM

**Pull request merged**